### DISCUSSION TITLE
Error while creating object in iCub_SIM using YARP bindings in MATLAB

### DISCUSSION THREAD

**Discussion by mshoaibb** [robotology/community#469] [Q&A] [answered]
i create the object in iCub_SIM using by yarp in terminal and got required results shown below.

![image](https://user-images.githubusercontent.com/75766541/103507423-efbf3c80-4e80-11eb-8372-e75af75b4648.png)

**### But when i try to do the same thing by using yarp bindings in matlab i get stuck. kindly help** 

**This is the code that i'm using in matlab**

```matlab
clear all;close all;
yarp.matlab.LoadYarp;
import yarp.Port;
import yarp.Bottle;
import yarp.BufferedPortImageRgb
import yarp.BufferedPortBottle;
portLeft=BufferedPortImageRgb;
portRight=BufferedPortImageRgb;
world=Port;
world.open('/matlab/world');
yarp.Network.connect('/icubSim/world','/matlab/world');
yarp.Network.init();
world.mk.box(0.3 0.3 0.03 0.3 0.2 1 1 0 0)
```

**and this is the error that i'm getting** 

```matlab
Error: File: untitled3.m Line: 13 Column: 18
Invalid expression. Check for missing multiplication operator, missing or unbalanced delimiters, or other syntax error. To
construct matrices, use brackets instead of parentheses.`
```

thanks.

**Comment by traversaro** [comment, marked as answer]
Hi @mshoaibb, the error refers to a specific line in your script, but given how you reported your script it is not clear which line is line 13. Can you please edit your message to contain the `untitled3.m` script correctly rendered, using for example the support for writing code blocks in Markdown, as documented in https://docs.github.com/en/free-pro-team@latest/github/writing-on-github/creating-and-highlighting-code-blocks ?

**Reply by pattacini** [reply]
The error seems to be generated by this line:
```matlab
world.mk.box(0.3 0.3 0.03 0.3 0.2 1 1 0 0)
```
Actually, MATLAB should expect a vector; therefore, the fix would be something like:
```matlab
world.mk.box([0.3 0.3 0.03 0.3 0.2 1 1 0 0]);
```

**Reply by traversaro** [reply]
I think that the problem is that the MATLAB object `world` is a [`yarp::os::Port`](http://www.yarp.it/git-master/classyarp_1_1os_1_1Port.html) instance, and so it does not have any `mk` attribute. I think to send the message in that form, it is necessary to first build a `yarp::os::Bottle` contain the message, and then send that Bottle over the port.

**Reply by mshoaibb** [reply]
thank you @traversaro and @pattacini  for your kind response.
but i'm still unable to make exact code that can give me required output.

kindly sir if anyone have documentation for "**Controlling iCub_SIM using yarp bindings in MATLAB**" then kindly share it with me.

**Reply by traversaro** [reply]
Hi @mshoaibb, I don't think anyone has that kind of documentation. If you are in doubt on how you need to prepare the `yarp::os::Bottle` for sending the message to iCub SIM, you can check this code that does it in C++: https://github.com/robotology/tool-affordances/blob/3e4cf9de6ee79968b9f2d7e5d0dee0a561c05950/modules/simtoolloader/src/simtoolloader.cpp#L927 , and try to convert it to MATLAB using the equivalent methods in the MATLAB bindings.
